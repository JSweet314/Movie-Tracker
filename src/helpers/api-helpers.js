import apiKey from '../private/apiKey.js';

export const getMovie = async () => {

  const response = await fetch(`https://api.themoviedb.org/3/movie/upcoming?api_key=${apiKey}&language=en-US&page=1`)
  const movie = await response.json()
  return movie.results

}