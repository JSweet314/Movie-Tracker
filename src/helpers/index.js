export const moviesWrangler = movieArray => {
  if (!Array.isArray(movieArray)) {
    return null;
  }
  return movieArray.map(movie => ({
    title: movie.title,
    releaseDate: movie.release_date,
    voteAverage: movie.vote_average,
    posterPath: movie.poster_path,
    overview: movie.overview,
    id: movie.id
  }));
};