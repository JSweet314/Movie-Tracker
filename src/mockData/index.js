export const mockMovies = [{
  "vote_count": 56,
  "id": 437670,
  "video": false,
  "vote_average": 6.2,
  "title": "Suck Me Shakespeer 3",
  "popularity": 28.024527,
  "poster_path": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
  "original_language": "de",
  "original_title": "Fack ju GÃ¶hte 3",
  "genre_ids": [35],
  "backdrop_path": "/a3Ipy3B4APTPx1ISkls95XSs5fU.jpg",
  "adult": false,
  "overview": "A comedy that follows an con who lands a position at a",
  "release_date": "2017-10-26"
}];

export const mockMovie = [{
  "movieId": 437670,
  "voteAverage": 6.2,
  "title": "Suck Me Shakespeer 3",
  "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
  "overview": "A comedy that follows an con who lands a position at a",
  "releaseDate": "2017-10-26"
}];

