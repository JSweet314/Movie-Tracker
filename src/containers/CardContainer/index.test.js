import { shallow } from 'enzyme';
import React from 'react';
import { CardContainer } from './index.js';
import * as mockData from '../../mockData/';
import * as helper from '../../helpers/';


describe('Card Container', () => {
  let wrapper;

  beforeEach(() => {
    const mockAddFavorite = jest.fn();
    const movies = helper.moviesWrangler(mockData.mockMovies);
    wrapper = shallow(
      <CardContainer movies={movies} handleClick={mockAddFavorite}/>
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});