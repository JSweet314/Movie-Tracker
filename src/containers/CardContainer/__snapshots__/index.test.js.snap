// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Container should match a snapshot for favorites route 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addFavorite={[Function]}
    displayedMovie=""
    favorites={
        Array [
            Object {
              "movieId": 437670,
              "overview": "A comedy that follows an con who lands a position at a",
              "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
              "releaseDate": "2017-10-26",
              "title": "Suck Me Shakespeer 3",
              "voteAverage": 6.2,
            },
          ]
    }
    hideMovieInfo={[Function]}
    logStatus={true}
    match={
        Object {
            "path": "/favorites",
          }
    }
    movies={
        Array [
            Object {
              "movieId": 437670,
              "overview": "A comedy that follows an con who lands a position at a",
              "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
              "releaseDate": "2017-10-26",
              "title": "Suck Me Shakespeer 3",
              "voteAverage": 6.2,
            },
          ]
    }
    removeFavorite={[Function]}
    showMovieInfoById={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          displayInfo=""
          favBtnClass=""
          handleFavorite={[Function]}
          handleInfoDisplay={[Function]}
          information={
                    Object {
                              "movieId": 437670,
                              "overview": "A comedy that follows an con who lands a position at a",
                              "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
                              "releaseDate": "2017-10-26",
                              "title": "Suck Me Shakespeer 3",
                              "voteAverage": 6.2,
                            }
          }
          isFavorited="favorite"
          logStatus={true}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "437670",
        "nodeType": "function",
        "props": Object {
          "displayInfo": "",
          "favBtnClass": "",
          "handleFavorite": [Function],
          "handleInfoDisplay": [Function],
          "information": Object {
            "movieId": 437670,
            "overview": "A comedy that follows an con who lands a position at a",
            "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
            "releaseDate": "2017-10-26",
            "title": "Suck Me Shakespeer 3",
            "voteAverage": 6.2,
          },
          "isFavorited": "favorite",
          "logStatus": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            displayInfo=""
            favBtnClass=""
            handleFavorite={[Function]}
            handleInfoDisplay={[Function]}
            information={
                        Object {
                                    "movieId": 437670,
                                    "overview": "A comedy that follows an con who lands a position at a",
                                    "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
                                    "releaseDate": "2017-10-26",
                                    "title": "Suck Me Shakespeer 3",
                                    "voteAverage": 6.2,
                                  }
            }
            isFavorited="favorite"
            logStatus={true}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "437670",
          "nodeType": "function",
          "props": Object {
            "displayInfo": "",
            "favBtnClass": "",
            "handleFavorite": [Function],
            "handleInfoDisplay": [Function],
            "information": Object {
              "movieId": 437670,
              "overview": "A comedy that follows an con who lands a position at a",
              "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
              "releaseDate": "2017-10-26",
              "title": "Suck Me Shakespeer 3",
              "voteAverage": 6.2,
            },
            "isFavorited": "favorite",
            "logStatus": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card Container should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addFavorite={[Function]}
    displayedMovie=""
    favorites={Array []}
    hideMovieInfo={[Function]}
    logStatus={true}
    match={
        Object {
            "path": "/",
          }
    }
    movies={
        Array [
            Object {
              "movieId": 437670,
              "overview": "A comedy that follows an con who lands a position at a",
              "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
              "releaseDate": "2017-10-26",
              "title": "Suck Me Shakespeer 3",
              "voteAverage": 6.2,
            },
          ]
    }
    removeFavorite={[Function]}
    showMovieInfoById={[Function]}
    user={
        Object {
            "email": "taco@taco",
            "id": 1,
            "name": "Taco",
            "password": "taco",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          displayInfo=""
          favBtnClass=""
          handleFavorite={[Function]}
          handleInfoDisplay={[Function]}
          information={
                    Object {
                              "movieId": 437670,
                              "overview": "A comedy that follows an con who lands a position at a",
                              "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
                              "releaseDate": "2017-10-26",
                              "title": "Suck Me Shakespeer 3",
                              "voteAverage": 6.2,
                            }
          }
          isFavorited=""
          logStatus={true}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "437670",
        "nodeType": "function",
        "props": Object {
          "displayInfo": "",
          "favBtnClass": "",
          "handleFavorite": [Function],
          "handleInfoDisplay": [Function],
          "information": Object {
            "movieId": 437670,
            "overview": "A comedy that follows an con who lands a position at a",
            "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
            "releaseDate": "2017-10-26",
            "title": "Suck Me Shakespeer 3",
            "voteAverage": 6.2,
          },
          "isFavorited": "",
          "logStatus": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            displayInfo=""
            favBtnClass=""
            handleFavorite={[Function]}
            handleInfoDisplay={[Function]}
            information={
                        Object {
                                    "movieId": 437670,
                                    "overview": "A comedy that follows an con who lands a position at a",
                                    "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
                                    "releaseDate": "2017-10-26",
                                    "title": "Suck Me Shakespeer 3",
                                    "voteAverage": 6.2,
                                  }
            }
            isFavorited=""
            logStatus={true}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "437670",
          "nodeType": "function",
          "props": Object {
            "displayInfo": "",
            "favBtnClass": "",
            "handleFavorite": [Function],
            "handleInfoDisplay": [Function],
            "information": Object {
              "movieId": 437670,
              "overview": "A comedy that follows an con who lands a position at a",
              "posterPath": "/cypnifmPJ5JbTjzpZI6MwJdvP7.jpg",
              "releaseDate": "2017-10-26",
              "title": "Suck Me Shakespeer 3",
              "voteAverage": 6.2,
            },
            "isFavorited": "",
            "logStatus": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
